src/classes/Indicators_Service.cls

```diff
@@ -1,67 +1,117 @@
-            }
-        } else if(config.Type__c == 'Blank') {
-            enabled = false;
-            display = false;
-        }
-        return new Indicator(config.Display_Value__c, enabled ? config.Enabled_Color__c : config.Disabled_Color__c, display, config.Hover_Text__c);
-    }
+		} else if(config.Type__c == 'Boolean') {
+			enabled = (Boolean)field;
+		} else if(config.Type__c == 'Is Not Null') {
+			if(String.isNotBlank(String.valueOf(field)))   //String.isNotBlank((String)field)
+				enabled = true;
+		} else if(config.Type__c == 'Greater Than') {
+			if(field != null && String.isNotBlank(config.Contains_Value__c)) {
+				enabled = (Decimal.valueOf(String.valueOf(field)) > Decimal.valueOf(config.Contains_Value__c));
+			}
+		} else if(config.Type__c == 'Less Than') {
+			if(field != null && String.isNotBlank(config.Contains_Value__c)) {
+				enabled = (Decimal.valueOf(String.valueOf(field)) < Decimal.valueOf(config.Contains_Value__c));
+			}
+		} else if(config.Type__c == 'Blank') {
+			enabled = false;
+			display = false;
+		}
 
-    public class Indicator {
-        public string Color {get;set;}
-        public string Value {get;set;}
-        public Boolean Display {get;set;}
-        public string Hover {get;set;}
+		// include null option as TEXT type and first to avoid null pointer exceptions and accomodate Blank indicator  
+		if (config.Display_Value__c == null || (config.Display_Value__c.length() <= 4 && !spriteSet.contains(config.Display_Value__c))) {
+			system.debug('Using text display for ' + config.Display_Value__c);
+			iType = 'TEXT';
+		} else if (Pattern.matches('custom[0-9]{1,3}',config.Display_Value__c)) {
+			system.debug('Custom sprite found for ' + config.Display_Value__c);
+			iType = 'CUSTOM_SPRITE';
+		} else if (spriteSet.contains(config.Display_Value__c)) {
+			system.debug('Standard sprite found for ' + config.Display_Value__c);
+			iType = 'STANDARD_SPRITE'; 
+		} else {
+			system.debug('No sprite found for ' + config.Display_Value__c + ' trying to add an error');
+			// ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.ERROR,'No valid icon found for Display Value ' + config.Display_Value__c + 'Check Indicator record ' + config.DeveloperName));
+			iType = 'TEXT';
+			throw new WeathervaneException('No valid icon found for Display Value ' + config.Display_Value__c + '. Check Indicator record ' + config.DeveloperName);
+		}
+		
+		return new Indicator(config.Display_Value__c, enabled ? config.Enabled_Color__c : config.Disabled_Color__c, display, config.Hover_Text__c, iType);
+	}
 
-        public Indicator(String value, String color, Boolean display, String hover) {
-            this.Value = value; this.Color = color; this.Display = display; this.Hover = hover;
-        }
-    }
+	public static Set<String> spriteSet {
+		get {
+			if (spriteSet == null) {
+/*				if (Cache.Org.contains('indicatorbar.partition1.spriteSet')) {
```

Comment:
I thought Platform Cache would be a cool way to make the list of icon names be a little quicker, but a) I had to request it in my dev org, and b) the managed package Platfom Cache scenario is sub-optimal for OSS projects anyway. From https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_cache_partition_setup.htm: 
> If you’re working with managed packages, we recommend using Branch Packaging Orgs to share a namespace across partitions. This feature lets you maintain multiple orgs or partitions as “branches” of your primary org. For information about Branch Packaging Orgs, contact Salesforce.